cli/core/docs/cli/cli-architecture.md

```diff
@@ -135,9 +171,19 @@ tanzu context use mgmt-cluster
 
 ## Target
 
-The Tanzu CLI supports two targets (context types): `kubernetes`, `mission-control`. This is currently backwards compatible, i.e., the plugins are still available at the root level. In addition to that, we also have contextual plugins grouped under the target.
+Target is a top level entity used to make the control plane, that a user is interacting against, more explicit in command invocations.
+This is done by creating a separate target specific command under root level command for Tanzu CLI. e.g. `tanzu <target>`
+
+The Tanzu CLI supports two targets: `kubernetes`, `mission-control`. This is currently backwards compatible, i.e., the plugins are still available at the root level.
+
+Target of a plugin is determined differently for Standalone Plugins and Context-Scoped plugins.
 
-List TKG workload clusters:
+For Standalone Plugins, the target is determined based on `target` field defined in `CLIPlugin` CR as part of the discovery API.
+
+For Context-scoped Plugins, the target is determined based on the `target` associated with Context itself.
+E.g. all plugins discovered through the Context `test-context` will have the same target that is associated with the `test-context`.
+
+List TKG workload clusters using `cluster` plugin associated with `kubernetes` target:
```

Comment:
This sentence needs an introduction.  How about changing things to:
"Using the `target` concept, we can differentiate between listing TKG workload clusters or TMC workload clusters.
To list TKG workload clusters using the TKG `cluster` plugin which is associated with the `kubernetes` target:"